### PR TITLE
detect and log invalid tf keys

### PR DESCRIPTION
**Pull request by apmoriarty**
Part of https://github.com/NationalSecurityAgency/datawave/pull/1632

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java [hunk 1]
```diff
@@ -18,7 +17,7 @@ public class TFKey {
     private String uidAndValue;
     
     private byte[] backing;
-    private ArrayList<Integer> nulls;
+    private final ArrayList<Integer> nulls = new ArrayList<>();
```
Comment (avgAGB) [inline warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java]: for readability would it be reasonable to change the naming here to nullValues or nullIntegers?

File: warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java [hunk 2]
```diff
@@ -38,11 +37,9 @@ public void parse(Key k) {
     }
     
     public String getDatatype() {
-        if (datatype == null) {
-            if (nulls.size() > 1) {
-                int stop = nulls.get(0);
-                datatype = new String(backing, 0, stop);
-            }
+        if (datatype == null && nulls.size() > 1) {
```
Comment (jschmidt10) [inline warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java]: This isn't a deal breaker but right now, we could have an invalid tf key but still allow `getDatatype()` to return something which may or may not be valid.
What do you think about all of the getXXX() methods instead doing:

```
if (datatype == null && isValid()) 
  // parse out datatype
```
We could do the same for uid and value, etc. That way, if it's invalid, we just return null for everything.

File: warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java]: `nulls` seems to be the convention when parsing a key, but perhaps `nullIndices` would be more precise....however typing that out I see why the decision was made for `nulls`

File: warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java (hunk 2 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java]: Simple enough -- done

File: warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java (hunk 1 above)
Comment (apmoriarty) [reply warehouse/query-core/src/main/java/datawave/core/iterators/key/TFKey.java]: Updated this variable name to `splits` with a comment

**Review by avgAGB — DISMISSED** [review]

**Review by avgAGB — APPROVED** [review]

**Review by jschmidt10 — APPROVED** [review]

**Pull request merged**